website/docs/r/service_account.html.markdown

```diff
@@ -0,0 +1,74 @@
+---
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_service_account"
+sidebar_current: "docs-vcd-resource-service_account"
+description: |-
+  Provides a resource to manage Service Accounts. Service Accounts can have defined roles
+  and act just like a VCD user. Service Accounts, when activated, provide one-time use
+  access tokens for authentication to the VCD API, during which a new access token is generated.
+---
+
+# vcd\_service\_account 
+
+Supported in provider *v3.10+* and VCD 10.4+.
+
+Provides a resource to manage Service Accounts. Service Accounts can have defined roles
+and act just like a VCD user. Service Accounts, when activated, provide one-time use
+access tokens for authentication to the VCD API, during which a new access token is generated.
+Explained in more detail [here].[service-accounts]
+
+## Example Usage 
+
+```hcl
+data "vcd_role" "vapp_author" {
+  org  = "my-org"
+  name = "vApp Author"
+}
+
+resource "vcd_service_account" "example_service" {
+  org  = "my-org"
+  name = "example"
+  role = data.vcd_role.vapp_author.id
+
+  software_id      = "12345678-1234-1234-1234-1234567890ab"
+  software_version = "1.0.0"
+  uri              = "example.com"
+
+  file_name = "example_service.json"
+
+  active = true
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful
+  when connected as sysadmin working across different organisations.
+* `name` - (Required) A unique name for the Service Account in an organisation.
+* `role` - (Required) ID of a Role.
+* `software_id` - (Required) UUID of the Service Account.
+* `software_version` - (Optional) Version of the service using the Service Account
+* `uri` - (Optional) URI of the service using the Service Account
+* `active` - (Required) Status of the Service Account. Can be set to `false` and back to `true` if
+  the access token was lost to get a new one.
+* `file_name` - (Optional) Required only when `active` is set to `true`. Contains the access token
```

Comment:
I think this is a great automation convenience layer we're adding and it deserves a cross reference to https://registry.terraform.io/providers/vmware/vcd/latest/docs#connecting-with-a-service-account-api-token

(meaning that we should reference this `file_name` can be used directly for provider auth and as well in the provider auth doc we should reference this resource.

Reply:
Same as in previous, I didn't want to make too much changes and will address this in a separate PR